### ISSUE TITLE
Feature Request: support for private requirements sources

### ISSUE THREAD

**Issue by issmirnov** [fubarhouse/ansible-role-tester#28]
First of all, thanks for a fantastic project! I had a question/feature request. Is it possible to get private requirements?

I'm setting up a personal CI/CD with Concourse, and I have about ~15 private git repos with my roles. When I test with vagrant, I simply specify the roles path to my project folder. When I add the roles to requirements.yml though, there's no SSH keys in the docker image.

I see two paths forward:

1. Inject SSH keys into the docker images somehow
2. Pass extra docker args to mount all my roles into some location in the container and add it as a roles path.

What are your thoughts? Does the docker image respect role_under_test/tests/ansible.cfg? Is it easy to inject SSH keys?

Advice appreciated, thanks!

**Comment by issmirnov** [comment]
So I realized that this is actually pretty simply to do - the code is incredibly clean and understandable, so I already have a patch up. I forked the repo and added a flag called `--extra-roles` that simply takes the host directory with roles and adds it as a volume to the docker container. 

Sample scenario:
```
$ cd /projects/ansible
$ ls
my_special_role dep1 dep2
$ cd my_special_role
$ ansible-role-tester full -d /etc/ansible/roles/$(basename $(pwd)) -t ubuntu1804 --playbook playbook.yml --extra-roles /projects/ansible
```

Since `my_special_role` depends on roles `dep1` and `dep2`, everything is mounted in the correct spot and the roles run.

As mentioned in the top comment, this could be a useful addition to the `requirements` mechanism for privately hosted roles or CI/CD environments where you are passing build artifacts.

Sending up a PR to see what Travis thinks of my changes.

**Comment by fubarhouse** [comment]
@issmirnov,

Thank you for your feedback! Rest assured I've worked with much more complicated projects and the simplicity found here is because of past experience. I hope this project serves you as it does me - and I've found it incredibly useful.

The _only_ thing I'd pick at, which could be adjusted by myself later is that this feature is only exposed to the `full` command. Would it be applicable to the `run` command if we're passing values when starting containers?

__Based upon your response, we can either tack that on, or I can merge and open an issue to look at it later.__

You've also requested that requirements be supported, and it actually is - but I require requirements so rarely that it probably isn't tested well enough.

Another cool little thing is running ansible roles on remote docker containers via inventory and/or manually specified. It is exciting what these features combined will be able to achieve.

Lastly, I'd like to thank you for making an effort to keep the simplicity consistent in your PR. I see a lot of complex code in my job, and I want you to know I do appreciate it.

**Comment by issmirnov** [comment]
@fubarhouse Thank you for such a wonderfully thoughtful reply. I'm honored to contribute to such a well maintained project!

I really liked your great suggestion on adding this to the `run` command. I think it's best to merge this PR as is, and I'll send a follow up. I found some minor bugs while adding that functionality, and I want to keep the git history clean and easy to follow.

I've already started utilizing the project in my development as well as my CI/CD pipelines, and found that it's a great alternative to my usual Vagrant cycle. Thank you for creating this.

I'm looking forward to seeing the enhancements to inventory management - this would make for some very interesting clusters and automation.
